xactserver/src/pg_dispatcher.rs

```diff
@@ -0,0 +1,59 @@
+use tokio_postgres::{connect, NoTls};
+use tokio::sync::mpsc;
+use bytes::{Buf, Bytes};
+
+pub struct PgDispatcher {
+    addr: String,
+}
+
+impl PgDispatcher {
+    pub fn new(addr: &str) -> PgDispatcher {
+        PgDispatcher {
+            addr: addr.to_owned(),
+        }
+    }
+
+    pub fn thread_main(&self, txn_rx: mpsc::Receiver<Bytes>) -> anyhow::Result<()> {
+        // obviously wrong cannot make a connection as soon as xactserver starts this thread
+        // must wait till first remote txn comes in
+        let rt = tokio::runtime::Builder::new_multi_thread()
+            .worker_threads(2)
+            .enable_all()
+            .build()
+            .unwrap();
+
+        rt.block_on(async move {
+            let mut txn_rx = txn_rx;
+            // Continuously listen for new tuple data from the postgres watcher
+            while let Some(mut buf) = txn_rx.recv().await {
+                let ip_port: Vec<&str> = self.addr.split(":").collect();
+                let conn_str = format!("host={} port={} user=postgres application_name=xactserver", ip_port[0], ip_port[1]);
+                
+                println!("connecting to local pg, conn_str: {}", conn_str);
+                let (client, conn) = connect(&conn_str, NoTls).await.unwrap();
```

Comment:
ideally, this should only be done once and retried if the connection is broken, but we can go with this for now. Maybe add a TODO so that we won't forget